cmake/sasMacros.cmake

```diff
@@ -79,21 +81,29 @@ macro(add_sas_test_new)
    endif(NOT ARG_CHECKERS AND NOT ARG_MODERNIZE)
 
    set(SAS_CMD "${CMAKE_BINARY_DIR}/scripts/compile")
+   set(FULL_FILENAME "${CMAKE_CURRENT_SOURCE_DIR}/${ARG_FILE}")
+   if(ARG_SA_CONFIGURATION_FILE)
+      set(CONFIGURATION_FILENAME "${CMAKE_CURRENT_SOURCE_DIR}/${ARG_SA_CONFIGURATION_FILE}")
+   endif()
+   if(ARG_WORKING_DIR)
+      set(WORKING_DIR "${CMAKE_SOURCE_DIR}/${ARG_WORKING_DIR}")
+   else()
+      set(WORKING_DIR "${CMAKE_CURRENT_LIST_DIR}")
+   endif()
 
-   set(FULL_FILENAME ${ARG_FILE})
-   # set(FULL_FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_FILE})
-   string(REPLACE "/" "." TEST_NAME ${FULL_FILENAME})
+   string(REPLACE "${CMAKE_SOURCE_DIR}/test/" "" TEST_NAME ${FULL_FILENAME})
+   string(REPLACE "/" "." TEST_NAME ${TEST_NAME})
+   string(REGEX REPLACE "\\.[^.]+$" "" TEST_NAME ${TEST_NAME})
 
-   # prepare environment variable for checkers:
-   foreach(CHECKER ${ARG_CHECKERS})
-      set(CHECKERS ${CHECKERS} "${CHECKER}:")
-   endforeach()
 
    add_test(NAME "${TEST_NAME}"
-     COMMAND ${SAS_CMD} --checkers=${CHECKERS} --modernize=${ARG_MODERNIZE} --cargs="${FULL_FILENAME} -c -std=c++11"
-     WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
+      COMMAND ${SAS_CMD} --checkers=${ARG_CHECKERS} --modernize=${ARG_MODERNIZE} --sa_configuration=${CONFIGURATION_FILENAME} "${ARG_FILE}" "-c" "-std=c++11"
```

Comment:
I noticed that this actually compiles the sources in the source directory. Since we now push through all compiler arguments, can we add a `-o` that ensures those temporary files are created in the build directory instead?